### DISCUSSION TITLE
AOT for Desktop

### DISCUSSION THREAD

**Discussion by Fancy2209** [airsdk/Adobe-Runtime-Support#3493] [Ideas]
HARMAN has said that they're pretty close to getting the AOT on Android, wich is great! But I'm curious, are there any plans for allowing the Halfmoon AOT Compiler/Runtime (Got that name from the AVMPlus source, I'm assuming the Halfmoon AOT is the one used at least) in Desktop Platforms? 
AS3 has always been considered slow compared to other languages (hell even under AVMPlus C/C++ was faster than AS3 normally, and the C code was actually transpiled to AS3 before compilation!), and I feel like it'd be great to see the AOT everywhere as an option at some point! 
Even if this was added I would still love if there were AS3 Revisions that would allow the AVM to run faster, as I'm sure the AOT has a lot of restrictions AVMPlus doesn't impose, making it hard to keep feature parity at 1:1

By the way, are there any news on the Android AOT? I'd love to know how the performance compares between AVMPlus and the Halfmoon AOT on Android!

**Comment by itlancer** [comment]
Also this feature request https://github.com/airsdk/Adobe-Runtime-Support/issues/1291

**Comment by bobaoapae** [comment]
I hope AOT for android come quickly.
For others platforms will be a great feature too.

**Comment by shaucky** [comment]
If AoT of other platforms can be realized, the performance optimization of AVM2 may be not important.
